corehq/apps/data_dictionary/tests/test_util.py

```diff
@@ -209,3 +247,25 @@ def test_is_case_type_or_prop_name_valid(self):
             self.assertTrue(is_case_type_or_prop_name_valid(valid_name))
         for invalid_name in invalid_names:
             self.assertFalse(is_case_type_or_prop_name_valid(invalid_name))
+
+
+@es_test(requires=[case_search_adapter], setup_class=True)
+class UsedPropsByCaseTypeTest(TestCase):
+
+    domain = uuid.uuid4().hex
+
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        case_block = CaseBlock(
+            case_id=uuid.uuid4().hex,
+            case_type='other-case-type',
+            case_name='Case A',
+            update={'other-prop': True},
+        )
+        case_search_es_setup(cls.domain, [case_block])
+
+    def test_get_used_props_by_case_type(self):
```

Comment:
should we add some more tests for this utility?
are there edge cases we should test for?
i see we tested for case property that should be present, how about the ones that should not be present?
may be we set up a couple of cases with overlapping and distinct case properties with some used/unused case properties, and match exactly the result of `get_used_props_by_case_type`?

Reply:
Good point, I definitely agree the more the merrier when it comes to tests. I can think of a few other things to test for, so will make a commit for some extra tests.

Reply:
I have added some additional checks in the test ([7cb0e51](https://github.com/dimagi/commcare-hq/pull/33956/commits/7cb0e516d480a515dcad1881219da0dde6b8fe51)). @mkangia Let me know if you think this looks good, or if we need more thorough testing.